pkg/validation/cluster.go

```diff
@@ -640,6 +640,46 @@ func validateNutanixCloudSpec(spec *kubermaticv1.NutanixCloudSpec) error {
 		}
 	}
 
+	if spec.PeUsername == "" {
+		if spec.CredentialsReference == nil {
+			return errors.New("no prism element username or credentials reference specified")
+		}
+
+		if err := kuberneteshelper.ValidateSecretKeySelector(spec.CredentialsReference, resources.NutanixPeUsername); err != nil {
+			return err
+		}
+	}
+
+	if spec.PePassword == "" {
+		if spec.CredentialsReference == nil {
+			return errors.New("no prism element password or credentials reference specified")
+		}
+
+		if err := kuberneteshelper.ValidateSecretKeySelector(spec.CredentialsReference, resources.NutanixPePassword); err != nil {
+			return err
+		}
+	}
+
+	if spec.PeEndpoint == "" {
+		if spec.CredentialsReference == nil {
+			return errors.New("no prism element endpoint or reference specified")
+		}
+
+		if err := kuberneteshelper.ValidateSecretKeySelector(spec.CredentialsReference, resources.NutanixPeEndpoint); err != nil {
+			return err
+		}
+	}
+
+	if spec.PePort == nil {
+		if spec.CredentialsReference == nil {
+			return errors.New("no prism element port or reference specified")
+		}
+
+		if err := kuberneteshelper.ValidateSecretKeySelector(spec.CredentialsReference, resources.NutanixPePort); err != nil {
+			return err
+		}
+	}
```

Comment:
Should be optional since we use a default value in the provider code, no? E2E also fails with:

```
[FAIL] - [Kubermatic] Create cluster
      {"Payload":{"error":{"details":null,"code":400,"message":"invalid cluster: cloud: Invalid value: \"\u003credacted\u003e\": no prism element port or reference specified"}}}
```